packages/grid/x-grid-modules/src/models/api/columnApi.ts

```diff
@@ -52,5 +52,5 @@ export interface ColumnApi {
    *
    * @param cols [[ColDef[]]]
    */
-  updateColumns: (cols: ColDef[]) => void;
+  updateColumns: (cols: ColDef[], resetState: boolean) => void;
```

Comment:
Second parameter should be optional and default to false
Also please don't forget to update the api comments above

Reply:
I don't understand why we need to reset the states of the columns. Does it mean that we will lose how they were sorted or filtered or resized?

Reply:
No because we just update the current state, keeping the same data, and just passing a new array

Reply:
I'm still confused. Maybe we should better document what `updateColumns` is for? Things start to be confusing where each column has its state: this sounds strange. The states I can imagine should likely be independent and controllable with their own prop, not tight to each column:

- columnWidths/defaultColumnWidths/onColumnWidthsChange e.g. https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/column-resizing/#controlled-mode
- sortBy/defaultSortBy/onSortByChange e.g. https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/sorting/#controlled-mode
- columnsOrder/defaultColumnsOrder/onColumnsOrderChange e.g. https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/column-reordering/#controlled-mode

Ok, maybe it's for internal optimizations. And then when it leaks to the public domain, it completely lost me.